File: Ej 16 (Cuantos Dias)/Java/CuantosDias.java
//DeathbatO

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Scanner;

public class CuantosDias{

    public static int calcularDiferenciaEnDias(String fecha1, String fecha2){
        try{
            SimpleDateFormat dateFormat = new SimpleDateFormat("dd/MM/yyyy");
            Date f1 = dateFormat.parse(fecha1);
            Date f2 = dateFormat.parse(fecha2);

            long diferenciaEnMillis = Math.abs(f1.getTime() - f2.getTime());
            long diferenciaEnDias = diferenciaEnMillis / (24 * 60 * 60 * 1000);

            return (int) diferenciaEnDias;
        } catch (ParseException e) {
            throw new IllegalArgumentException("Una o ambas fechas no son válidas", e);
        }
    }

    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);

        System.out.print("Digite una fecha en formato dd/MM/yyyy: ");
        String fecha1 = sc.next();

        System.out.print("Digite una fecha en formato dd/MM/yyyy: ");
        String fecha2 = sc.next();

        int diferencia = calcularDiferenciaEnDias(fecha1, fecha2);
        System.out.println("La diferencia en días entre " + fecha1 + " y " + fecha2 + " es: " + diferencia + " días.");
        sc.close();
    }
}

File: Ej 16 (Cuantos Dias)/Python/CuantosDias.py
## DeathbatO

from datetime import datetime as dt

def CalcularDifDias(fecha1, fecha2):

    try:

        f1 = dt.strptime(fecha1, '%d/%m/%Y')
        f2 = dt.strptime(fecha2, '%d/%m/%Y')
        dif = abs((f1-f2).days)
        return dif
    
    except ValueError as e:

        raise ValueError("Una o ambas fechas no son validas") from e
    
fe1 = input("Digite una fecha en formato dd/mm/YY: ")
fe2 = input("Digite una fecha en formato dd/mm/YY: ")

dif = CalcularDifDias(fe1,fe2)
print(f"La diferencia en días entre {fe1} y {fe2} es: {dif} días.")
